### ISSUE TITLE
Black settings

### ISSUE THREAD

**Issue by roberto-arista** [typemytype/drawbot#550]
Hey!
I see black listed in the dependencies, but I cannot find the settings in the repo. Are you using it to format the code?
I'd like to adhere to the repository customs : )

**Comment by justvanrossum** [comment]
Black is a dependency for the "Format Code" menu under "Python". Unfortunately, the code base itself doesn't use it...

**Comment by roberto-arista** [comment]
Is there any interest in settings it up?

**Comment by justvanrossum** [comment]
It will be a huge commit. I’d be in favor but only if @typemytype is comfortable with it, too.

If there are any unmerged prs they should be merged first.

Are you suggesting to set up pre-commit?

**Comment by typemytype** [comment]
the app requirements and packages requirements could be separated. 

black is only used inside the app.

add an `app-requirements.txt` and install them here 
https://github.com/typemytype/drawbot/blob/master/.github/workflows/build.yml#L51

app requirements could be: (out of my head...) 
```
black
corefoundationasyncio
py2app
pygments
pip```

**Comment by justvanrossum** [comment]
Maybe take the opportunity to rename the req files with "requirements" at the front, so they naturally group together:

- `requirements.txt`
- `requirements-rtd.txt`
- `requirements-test.txt` (perhaps `requirements-dev.txt`?)
- `requirements-app.txt`

?

**Comment by roberto-arista** [comment]
> Are you suggesting to set up pre-commit?

I generally store the black settings in a pyproject.toml file. If there's interest, we could write them somewhere that could be picked up by different code editors...

**Comment by justvanrossum** [comment]
Summarizing:
- black is currently used to format user code via the "Format Code" menu
- this uses the default black settings
- we currently do not use black when editing the DrawBot code base

Conclusion:
- we don't need black settings to be stored anywhere

The question remains:
- Should we consider using black on the DrawBot code base itself? I would be in favor, but it is ultimately @typemytype's decision. This should be a new issue, though, so I'm closing this now.
